client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,469 @@
+                current_blockheight.store(blockheight, Ordering::Relaxed);
+            }
+            tokio::time::sleep(BLOCKHASH_REFRESH_RATE).await;
+        }
+    })
+}
+
+fn create_transaction_confirmation_task(
+    rpc_client: Arc<RpcClient>,
+    current_blockheight: Arc<AtomicU64>,
+    transaction_map: Arc<DashMap<Signature, TransactionData>>,
+    errors_map: Arc<DashMap<usize, TransactionError>>,
+    confirmed_transactions: Arc<AtomicU32>,
+) -> JoinHandle<()> {
+    tokio::spawn(async move {
+        // we also check last time all transaction that have just expired between two checks
+        let mut last_block_height = current_blockheight.load(Ordering::Relaxed);
+
+        loop {
+            if !transaction_map.is_empty() {
+                let current_blockheight = current_blockheight.load(Ordering::Relaxed);
+                let transactions_to_verify: Vec<Signature> = transaction_map
+                    .iter()
+                    .filter(|x| {
+                        // all transactions that are not expired
+                        let is_not_expired = current_blockheight <= x.last_valid_blockheight;
+                        // all transaction that expired between last and current check
+                        let is_recently_expired = last_block_height <= x.last_valid_blockheight
+                            && current_blockheight > x.last_valid_blockheight;
+                        is_not_expired || is_recently_expired
+                    })
+                    .map(|x| *x.key())
+                    .collect();
+                for signatures in
+                    transactions_to_verify.chunks(MAX_GET_SIGNATURE_STATUSES_QUERY_ITEMS)
+                {
+                    if let Ok(result) = rpc_client.get_signature_statuses(signatures).await {
+                        let statuses = result.value;
+                        for (signature, status) in signatures.iter().zip(statuses.into_iter()) {
+                            status
+                                .filter(|status| status
+                                    .satisfies_commitment(rpc_client.commitment()
+                                )
+                                .and_then(|status| transaction_map
+                                    .remove(signature)
+                                    .map(|(_, data)| (status, data))
+                                )
+                                .and_then(|(status, data)| {
+                                    confirmed_transactions.fetch_add(1, Ordering::Relaxed);
+                                    if let Some(error) = status.err {
+                                        errors_map.insert(data.index, error);
+                                    }
+                                });
+                        }
+                    }
+                }
+
+                last_block_height = current_blockheight;
+            }
+            tokio::time::sleep(Duration::from_millis(100)).await;
```

Comment:
I think it should be at least one block and in any case I would use slot as a unit for this